### PR TITLE
chore: Prepare v0.16.0 release

### PR DESCRIPTION
**Pull request by pront**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,28 @@
 
 <!-- changelog start -->
 
+## [0.15.0 (2024-06-06)]
```
Comment (neuronull) [inline CHANGELOG.md:7]:
```suggestion
## [0.16.0 (2024-06-06)]
```

Comment (neuronull) [reply CHANGELOG.md]: 🤔  was this auto generated ?

Comment (pront) [reply CHANGELOG.md]: Good catch, does it read the version from the Cargo.toml at the time the script is ran?

**Review by neuronull — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (neuronull) [reply CHANGELOG.md]: ah it does yeah. I think this implies the cargo.toml was bumped after the changelog generation ?

Comment (pront) [reply CHANGELOG.md]: Correct, at some point the publish script can take care of this to avoid manual errors.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,28 @@
...
+### Breaking Changes & Upgrade Guide
+
+- The deprecated coalesce paths (i.e. `(field1|field2)`) feature is now removed. (https://github.com/vectordotdev/vrl/pull/836)
+
+### New Features
+
+- Added experimental `dns_lookup` function. It should be used with caution, since it involves network
+  calls and is therefore very slow.
+
+  authors: esensar (https://github.com/vectordotdev/vrl/pull/764)
```
Comment (jszwedko) [inline CHANGELOG.md:19]: Should this line be here? cc/ @neuronull

Comment (neuronull) [reply CHANGELOG.md:19]: FYI I followed up with a manual removal of these for now.

**Pull request merged**